edgelet/iotedge/src/system.rs

```diff
@@ -13,27 +13,30 @@ use crate::error::{Error, ErrorKind};
 
 lazy_static! {
     static ref IOTEDGED: ServiceDefinition = {
-        // Use the presence of IOTEDGE_HOST to infer whether this is being built for CentOS 7 or not.
-        // CentOS 7 doesn't use socket activation.
-        let sockets: &'static [&'static str] = option_env!("IOTEDGE_HOST").map_or(
-            &["aziot-edged.mgmt.socket", "aziot-edged.workload.socket"],
-            |_host| &[],
-        );
+        // If IOTEDGE_LISTEN_MANAGEMENT_URI isn't set at compile-time, assume socket activation is being used.
+        //
+        // This doesn't matter for release builds since those always have IOTEDGE_LISTEN_MANAGEMENT_URI set.
+        // It's only useful for developers' builds.
+        let uses_socket_activation = option_env!("IOTEDGE_LISTEN_MANAGEMENT_URI").map_or(true, |value| value.starts_with("fd://"));
```

Comment:
It is best to be explicit here and check for cfg_attr(debug_assertions)

Reply:
It's not about debug-vs-not. It's about `make rpm` / `make deb` vs not.

Reply:
I get it now. The comment threw me off as it read "release builds" (I suggest "package build") and "developers' build" (I suggest "non-RPM/DEB builds")

Reply:
Oh right. Fixed.